### ISSUE TITLE
Hue Level and Color change doesn't notify Domo / MQTT

### ISSUE THREAD

**Issue by jonferreira** [domoticz/domoticz#2281]
If you change Level on the Hue App / Switch MQTT isn't notified and when you change Color with the Hue App neither Domoticz or MQTT is notified. 

In my case I've changed Color quite a few times with the Hue App in the past few minutes and the Color variable is stuck at "{"b":81,"cw":0,"g":255,"m":3,"r":29,"t":0,"ww":0}" (last color change I did with domo)

**Comment by jonferreira** [comment]
#2273

**Comment by jonferreira** [comment]
Further tests with http://montnemery.com/domoticz_hue/linux_armv7l/001/domoticz_linux_armv7l.tgz

**No MQTT update when turning on/off in an external app**
Sorted

```
2018-04-04 13:09:09.324 domoticz/out {
2018-04-04 13:09:09.336    "Battery" : 255,
2018-04-04 13:09:09.348    "Color" : "",
2018-04-04 13:09:09.360    "Level" : 100,
2018-04-04 13:09:09.372    "RSSI" : 12,
2018-04-04 13:09:09.385    "description" : "",
2018-04-04 13:09:09.396    "dtype" : "Color Switch",
2018-04-04 13:09:09.408    "id" : "00000002",
2018-04-04 13:09:09.421    "idx" : 2,
2018-04-04 13:09:09.433    "name" : "Hue color spot 1",
2018-04-04 13:09:09.450    "nvalue" : 0,
2018-04-04 13:09:09.462    "stype" : "RGBWW",
2018-04-04 13:09:09.474    "svalue1" : "100",
2018-04-04 13:09:09.486    "switchType" : "Dimmer",
2018-04-04 13:09:09.498    "unit" : 1
2018-04-04 13:09:09.509 }
2018-04-04 13:09:09.521
2018-04-04 13:09:09.534 domoticz/out {
2018-04-04 13:09:09.546    "Battery" : 255,
2018-04-04 13:09:09.557    "RSSI" : 12,
2018-04-04 13:09:09.569    "description" : "",
2018-04-04 13:09:09.581    "dtype" : "Color Switch",
2018-04-04 13:09:09.593    "id" : "000003EB",
2018-04-04 13:09:09.605    "idx" : 54,
2018-04-04 13:09:09.617    "name" : "Group Living Room",
2018-04-04 13:09:09.629    "nvalue" : 0,
2018-04-04 13:09:09.640    "stype" : "RGBWW",
2018-04-04 13:09:09.652    "svalue1" : "100",
2018-04-04 13:09:09.664    "switchType" : "On/Off",
2018-04-04 13:09:09.675    "unit" : 1
2018-04-04 13:09:09.687 }
2018-04-04 13:09:09.699
```
**Web UI does not auto refresh when turning on/off in an external app**
Sorted

**No MQTT update when Level is changed in an external app**

Scenario:

Hue color spot 1 bulb in Hues' Living Room group
In Hue app open group Living Room and dim Hue color spot 1 to 50%

Result:

Domoticz UI is updated
MQTT isn't notified about the Level change **but** it notifies about a change on Living Room Group 

(note svalue1)
```
2018-04-04 13:12:00.462 domoticz/out {
2018-04-04 13:12:00.474    "Battery" : 255,
2018-04-04 13:12:00.487    "RSSI" : 12,
2018-04-04 13:12:00.499    "description" : "",
2018-04-04 13:12:00.511    "dtype" : "Color Switch",
2018-04-04 13:12:00.523    "id" : "000003EB",
2018-04-04 13:12:00.535    "idx" : 54,
2018-04-04 13:12:00.547    "name" : "Group Living Room",
2018-04-04 13:12:00.559    "nvalue" : 1,
2018-04-04 13:12:00.571    "stype" : "RGBWW",
2018-04-04 13:12:00.583    "svalue1" : "50",
2018-04-04 13:12:00.596    "switchType" : "On/Off",
2018-04-04 13:12:00.607    "unit" : 1
2018-04-04 13:12:00.620 }
2018-04-04 13:12:00.632
```

Expected:

MQTT should also get a notification with

```
2018-04-04 13:10:19.884 domoticz/out {
2018-04-04 13:10:19.896    "Battery" : 255,
2018-04-04 13:10:19.907    "Color" : "",
2018-04-04 13:10:19.919    "Level" : 50,
2018-04-04 13:10:19.931    "RSSI" : 12,
2018-04-04 13:10:19.943    "description" : "",
2018-04-04 13:10:19.955    "dtype" : "Color Switch",
2018-04-04 13:10:19.967    "id" : "00000002",
2018-04-04 13:10:19.979    "idx" : 2,
2018-04-04 13:10:19.991    "name" : "Hue color spot 1",
2018-04-04 13:10:20.003    "nvalue" : 1,
2018-04-04 13:10:20.015    "stype" : "RGBWW",
2018-04-04 13:10:20.027    "svalue1" : "50",
2018-04-04 13:10:20.040    "switchType" : "Dimmer",
2018-04-04 13:10:20.052    "unit" : 1
2018-04-04 13:10:20.065 }
```

**No Domoticz or MQTT update when Color is changed in an external app**

Simply change color on Hue App and "nothing" happens on Domoticz. 

```
2018-04-04 13:15:21.607  PhilipsHue::InsertUpdateSwitch NodeID: 00000002, LType: 5, bIsOn: 1
2018-04-04 13:15:21.608  SQLQ query : SELECT nValue, SubType, ID FROM DeviceStatus WHERE (HardwareID==2) AND (Unit==1) AND (Type==241) AND (DeviceID=='00000002')
2018-04-04 13:15:21.609  SQLR result: 1;4;2;
2018-04-04 13:15:21.609  SQLR result: 1;4;52;
2018-04-04 13:15:21.610  --PhilipsHue::InsertUpdateSwitch !result.empty()
2018-04-04 13:15:21.610  --PhilipsHue::InsertUpdateSwitch updating nValue etc.
2018-04-04 13:15:21.612  SQLQ query : UPDATE DeviceStatus SET nValue=1, sValue='253;64668', LastLevel = 100, LastUpdate='2018-04-04 13:15:21' WHERE(HardwareID == 2) AND (DeviceID == '00000002')
2018-04-04 13:15:21.613  PhilipsHue::InsertUpdateSwitch NodeID: 000003eb, LType: 5, bIsOn: 1
2018-04-04 13:15:21.614  SQLQ query : SELECT nValue, SubType, ID FROM DeviceStatus WHERE (HardwareID==2) AND (Unit==1) AND (Type==241) AND (DeviceID=='000003eb')
2018-04-04 13:15:21.615  --PhilipsHue::InsertUpdateSwitch calling PushAndWaitRxMessage
2018-04-04 13:15:21.616  MAIN ProcessRX Msg 1BF10400EB030000010100006400000000000000000000000000AF70
2018-04-04 13:15:21.616  SQLQ query : SELECT ID,Name, Used, SwitchType, nValue, sValue, LastUpdate, Options FROM DeviceStatus WHERE (HardwareID=2 AND DeviceID='000003EB' AND Unit=1 AND Type=241 AND SubType=4)
2018-04-04 13:15:21.616  SQLR result: 54;Group Living Room;0;0;1;100;2018-04-04 13:15:01;;
2018-04-04 13:15:21.617  SQLQ query : UPDATE DeviceStatus SET SignalLevel=12, BatteryLevel=255, nValue=1, sValue='100', LastUpdate='2018-04-04 13:15:21' WHERE (ID = 54)
2018-04-04 13:15:21.617  SQLQ query : INSERT INTO LightingLog (DeviceRowID, nValue, sValue) VALUES ('54', '1', '100')
2018-04-04 13:15:21.618  SQLQ query : SELECT ID FROM DeviceStatus WHERE (HardwareID=2 AND DeviceID='000003EB' AND Unit=1 AND Type=241 AND SubType=4)
2018-04-04 13:15:21.618  SQLR result: 54;
2018-04-04 13:15:21.618  SQLQ query : SELECT A.ParentID, B.Name, B.HardwareID, B.[Type], B.[SubType], B.Unit FROM LightSubDevices as A, DeviceStatus as B WHERE (A.DeviceRowID=='54') AND (A.DeviceRowID!=A.ParentID) AND (B.[ID] == A.ParentID)
2018-04-04 13:15:21.619  SQLQ query : SELECT a.DeviceRowID, b.Type FROM LightSubDevices a, DeviceStatus b WHERE (a.ParentID=='54') AND (b.ID == a.DeviceRowID) AND (a.DeviceRowID!=a.ParentID)
2018-04-04 13:15:21.619  SQLQ query : SELECT ID, Activators, SceneType FROM Scenes WHERE (Activators!='')
2018-04-04 13:15:21.619  SQLQ query : UPDATE DeviceStatus SET BatteryLevel=255 WHERE (ID==54)
2018-04-04 13:15:21.619  (Hue) Color Switch (Group Living Room)
2018-04-04 13:15:21.620  SQLQ query : SELECT DeviceID, Unit, Name, [Type], SubType, nValue, sValue, SwitchType, SignalLevel, BatteryLevel, Options, Description, LastLevel, Color FROM DeviceStatus WHERE (HardwareID==2) AND (ID==54)
2018-04-04 13:15:21.620  SQLR result: 000003EB;1;Group Living Room;241;4;1;100;0;12;255;;;0;;
2018-04-04 13:15:21.620  MQTT: Publish: domoticz/out ({
   "Battery" : 255,
   "RSSI" : 12,
   "description" : "",
   "dtype" : "Color Switch",
   "id" : "000003EB",
   "idx" : 54,
   "name" : "Group Living Room",
   "nvalue" : 1,
   "stype" : "RGBWW",
   "svalue1" : "100",
   "switchType" : "On/Off",
   "unit" : 1
}
)
2018-04-04 13:15:21.622  --PhilipsHue::InsertUpdateSwitch set Name=Group Living Room
2018-04-04 13:15:21.624  SQLQ query : UPDATE DeviceStatus SET Name='Group Living Room', SwitchType=7, nValue=1, sValue='253;64668', LastLevel=100 WHERE(HardwareID == 2) AND (DeviceID == '000003eb')
```

Similar to the above MQTT is notified of a "group event" although with no relevant information regarding the color
```

2018-04-04 13:15:21.633 domoticz/out {
2018-04-04 13:15:21.645    "Battery" : 255,
2018-04-04 13:15:21.656    "RSSI" : 12,
2018-04-04 13:15:21.668    "description" : "",
2018-04-04 13:15:21.680    "dtype" : "Color Switch",
2018-04-04 13:15:21.692    "id" : "000003EB",
2018-04-04 13:15:21.704    "idx" : 54,
2018-04-04 13:15:21.716    "name" : "Group Living Room",
2018-04-04 13:15:21.727    "nvalue" : 1,
2018-04-04 13:15:21.739    "stype" : "RGBWW",
2018-04-04 13:15:21.750    "svalue1" : "100",
2018-04-04 13:15:21.762    "switchType" : "On/Off",
2018-04-04 13:15:21.774    "unit" : 1
2018-04-04 13:15:21.786 }
2018-04-04 13:15:21.797

```

Hope you can make sense out of this

**Comment by emontnemery** [comment]
Thanks for the detailed test report!

**Regarding no MQTT update for Hue color spot 1:**
The reason there is no MQTT message for the light itself is because the light (according to domoticz) was already switched on, so no need to send an update.
This could be because:
 - You were switching on and off too quickly in the external app and Domoticz had no chance to refresh the state. 
By default, Domoticz will poll the hue bridge for status every 10 s, you can shorten that time on the HW settings page, or just wait longer when testing
- There is an issue with multiple devices: the database query finds the "Hue color spot 1" as both idx 2 and 52. Please delete the one with idx 52, I think it was added earlier by the broken device upgrade code.

It seems clear why the groups and scenes are constantly renamed by the way, it's due to Philips Hue (incorrectly) using lower case for the ID whereas rest of Domoticz is using upper case.
Did you not see that issue on older beta, it happened first after #2229 was merged??

**Regarding no valid Color :**
- Let's fix this once the status update is working reliably..

**Comment by jonferreira** [comment]
OK I've just tested this on the latest stable release and indeed the behaviour is similar, i.e. no MQTT notification on color or level update plus color change on the Hue app doesn't update Domoticz - so not exactly a recent bug.

Regarding your comments, I've been giving it plenty of time before doing any update. I have the verbose log running on a screen session and only do any action before I see that domoticz have polled the Hue Bridge at least once, i.e. give or take 10 seconds.

I've deleted 52 now but saw no changes - really sounds like it's something that was never implemented I reckon :(

**Comment by emontnemery** [comment]
OK, and again:
The reason there is no MQTT message for the light itself is because the light (according to Domoticz) was already switched on, so no need to send an update.
This means:
- If you switch on/off in the external app, you **should** get notifications
- If you change brightness in the external app, there will be no notification (brightness will only be updated when the external app makes light go from off to on)
- If you change color in the external app, there will be no notification (it's not implemented)

If you really don't get notifications when switching on/off, it would be very helpful with a longer log where you switch on in the external app, wait for Domoticz to poll, switch off in the external app, wait for Domoticz to poll and repeat this a few times.

I want to make sure on/off notification issue is clarified before starting work on brightness/color notifications.

**Comment by jonferreira** [comment]
Yeah, On / Off status update is sorted mate as per - https://github.com/domoticz/domoticz/issues/2281#issuecomment-378579934

It's just the brightness / colour MQTT notifications now :)

>>>>>> brightness will only be updated when the external app makes light go from off to on
Yes, I understand the "why" although it would be a good improvement :)

**Comment by emontnemery** [comment]
@jonferreira Thanks, I got a bit confused there.. I'll go ahead with fixing the remaining issues with brightness and color change in external app.

**Comment by jonferreira** [comment]
sounds great, thanks for the effort!

just as a side note, a quick question: why can't one change any of the Hues' devices icon? I mean you can select an icon when you "edit" the device but it doesn't do anything on the dashboard...just curious really

**Comment by emontnemery** [comment]
Please try this build: http://montnemery.com/domoticz_hue/linux_armv7l/002/domoticz_linux_armv7l.tgz
Changes:
- Names of scenes and groups should no longer be reverted
- Web UI and MQTT should be updated when you use app to set brightness, color or color temperature

Icon is hardcoded in the Domoticz WebUI for dimmers and color switches.
I guess the icon selector should be hidden for such devices, please create a new issue if you want that fixed..

**Comment by jonferreira** [comment]
OK so good news and bad news.

Changing Level does notify MQTT with the Level change - perfect

Changing Color now send an MQTT notification but with the Color field empty

```

2018-04-06 12:13:21.779 domoticz/out {
2018-04-06 12:13:21.799    "Battery" : 255,
2018-04-06 12:13:21.811    "Color" : "",
2018-04-06 12:13:21.823    "Level" : 100,
2018-04-06 12:13:21.836    "RSSI" : 12,
2018-04-06 12:13:21.848    "description" : "",
2018-04-06 12:13:21.860    "dtype" : "Color Switch",
2018-04-06 12:13:21.872    "id" : "00000002",
2018-04-06 12:13:21.884    "idx" : 2,
2018-04-06 12:13:21.897    "name" : "Hue color spot 1",
2018-04-06 12:13:21.909    "nvalue" : 10,
2018-04-06 12:13:21.921    "stype" : "RGBWW",
2018-04-06 12:13:21.933    "svalue1" : "100",
2018-04-06 12:13:21.945    "switchType" : "Dimmer",
2018-04-06 12:13:21.957    "unit" : 1
2018-04-06 12:13:21.969 }
2018-04-06 12:13:21.981
2018-04-06 12:13:21.994 domoticz/out {
2018-04-06 12:13:22.005    "Battery" : 255,
2018-04-06 12:13:22.017    "RSSI" : 12,
2018-04-06 12:13:22.029    "description" : "",
2018-04-06 12:13:22.041    "dtype" : "Color Switch",
2018-04-06 12:13:22.053    "id" : "000003EB",
2018-04-06 12:13:22.065    "idx" : 8,
2018-04-06 12:13:22.077    "name" : "Group Living Room",
2018-04-06 12:13:22.089    "nvalue" : 10,
2018-04-06 12:13:22.101    "stype" : "RGBWW",
2018-04-06 12:13:22.113    "svalue1" : "100",
2018-04-06 12:13:22.125    "switchType" : "On/Off",
2018-04-06 12:13:22.137    "unit" : 1
2018-04-06 12:13:22.149 }
2018-04-06 12:13:22.161
```

As far as I can see the color picker on domoticz never gets updated either

![image](https://user-images.githubusercontent.com/7714292/38418376-1725dd06-3994-11e8-910c-61ad4b8269f7.png)

(bulb is green right now)

**Comment by emontnemery** [comment]
Ok!
Can you please include the logs starting with PhilipsHue::InsertUpdateSwitch?

**Comment by jonferreira** [comment]
I reckon what you're looking for is

2018-04-06 12:13:21.764  --PhilipsHue::InsertUpdateSwitch updating nValue etc.
2018-04-06 12:13:21.766  PhilipsHue::InsertUpdateSwitch NodeID: 000003eb, LType: 5, bIsOn: 1
2018-04-06 12:13:21.768  --PhilipsHue::InsertUpdateSwitch !result.empty()
2018-04-06 12:13:21.768  --PhilipsHue::InsertUpdateSwitch setting color to {INVALID}
2018-04-06 12:13:21.768  --PhilipsHue::InsertUpdateSwitch calling PushAndWaitRxMessage

Let meknow if you want the full log

**Comment by emontnemery** [comment]
I suspect it's because your light is updating its color in the "xy" format, but let's see..
Try this build: http://montnemery.com/domoticz_hue/linux_armv7l/003/domoticz_linux_armv7l.tgz
There should now be some addtional print like:
`Philips Hue: colormode`

**Comment by jonferreira** [comment]
yes, I get loads of messages like that everytime the bridge is polled

2018-04-06 16:22:07.953  Philips Hue: colormode 'xy'
2018-04-06 16:22:07.953  Philips Hue: colormode 'xy'
2018-04-06 16:22:07.953  Philips Hue: colormode 'xy'
2018-04-06 16:22:07.953  Philips Hue: colormode 'hs'
2018-04-06 16:22:07.954  Philips Hue: colormode 'xy'
2018-04-06 16:22:07.954  Philips Hue: colormode 'xy'
2018-04-06 16:22:07.954  Philips Hue: colormode 'xy'
2018-04-06 16:22:07.954  Philips Hue: colormode 'hs'
2018-04-06 16:22:18.561  Philips Hue: colormode 'xy'
2018-04-06 16:22:18.561  Philips Hue: colormode 'xy'
2018-04-06 16:22:18.561  Philips Hue: colormode 'xy'
2018-04-06 16:22:18.562  Philips Hue: colormode 'hs'
2018-04-06 16:22:18.562  Philips Hue: colormode 'xy'
2018-04-06 16:22:18.562  Philips Hue: colormode 'xy'
2018-04-06 16:22:18.562  Philips Hue: colormode 'xy'
2018-04-06 16:22:18.562  Philips Hue: colormode 'hs'

**Comment by jonferreira** [comment]
2018-04-06 16:23:33.179  Philips Hue: colormode 'xy'
2018-04-06 16:23:33.180  PhilipsHue::InsertUpdateSwitch NodeID: 000003eb, LType: 5, bIsOn: 1
2018-04-06 16:23:33.180  SQLQ query : SELECT nValue, LastLevel, Color, SubType, ID FROM DeviceStatus WHERE (HardwareID==2) AND (Unit==1) AND (Type==241) AND (DeviceID=='000003EB')
2018-04-06 16:23:33.180  SQLR result: 10;100;;4;8;
2018-04-06 16:23:33.180  --PhilipsHue::InsertUpdateSwitch !result.empty()
2018-04-06 16:23:33.181  --PhilipsHue::InsertUpdateSwitch setting color to {INVALID}
2018-04-06 16:23:33.181  --PhilipsHue::InsertUpdateSwitch calling PushAndWaitRxMessage
2018-04-06 16:23:33.181  MAIN ProcessRX Msg 1BF10470EB030000010A000064000000000000000000000000009F7

**Comment by emontnemery** [comment]
Thanks!
I need to add translation from Hue's xy colormode to RGB also.

**Comment by jonferreira** [comment]
Anything you need just let me know

**Comment by emontnemery** [comment]
Please test this build, now xy color should be converted to RGB:
http://montnemery.com/domoticz_hue/linux_armv7l/005/domoticz_linux_armv7l.tgz

By the way, can you confirm you get the correct number of events (1) when switching on/off, changing brightness, changing color, changing color temperature both from Domoticz WebUI and from external app?

**Comment by jonferreira** [comment]
OK So far it all seems to work fine with both Level and Color changes being published to MQTT and Domo UI updating automatically - awesome!

There was this one time where I did noticed that I got 2 updates per action but I could never replicated again so I'll assume it was me who did something wrong.

I'll assume you'll eventually push this to the Beta branch?

**Comment by emontnemery** [comment]
Great!
Yeah, I'll submit a pull request with the changes soon.

**Comment by jonferreira** [comment]
Cool.

I'll open another issue regarding the hardcoded icon - https://github.com/domoticz/domoticz/issues/2281#issuecomment-379063994

Hopefully it's easy enough to just use the custom icon like every other device in domoticz

**Comment by emontnemery** [comment]
There seemed to be a few unecessary database accesses from Domoticz Philips Hue code, I tried to clean that up.

Please test this version too, including discovery of some lights, some groups and some scenes:
http://montnemery.com/domoticz_hue/linux_armv7l/007/domoticz_linux_armv7l.tgz

Then it should be ready for merge :)

**Comment by jonferreira** [comment]
OK now it seems I get double MQTT notifications everytime color is changed.

Here's some examples

```
2018-04-10 09:46:16.059 domoticz/out {
2018-04-10 09:46:16.076    "Battery" : 255,
2018-04-10 09:46:16.088    "Color" : "{\"b\":31,\"cw\":0,\"g\":66,\"m\":3,\"r\":254,\"t\":0,\"ww\":0}",
2018-04-10 09:46:16.100    "Level" : 98,
2018-04-10 09:46:16.113    "RSSI" : 12,
2018-04-10 09:46:16.124    "description" : "",
2018-04-10 09:46:16.137    "dtype" : "Color Switch",
2018-04-10 09:46:16.148    "id" : "00000002",
2018-04-10 09:46:16.160    "idx" : 2,
2018-04-10 09:46:16.172    "name" : "Hue color spot 1",
2018-04-10 09:46:16.184    "nvalue" : 10,
2018-04-10 09:46:16.196    "stype" : "RGBWW",
2018-04-10 09:46:16.208    "svalue1" : "98",
2018-04-10 09:46:16.220    "switchType" : "Dimmer",
2018-04-10 09:46:16.232    "unit" : 1
2018-04-10 09:46:16.244 }
2018-04-10 09:46:16.256
2018-04-10 09:46:16.267 domoticz/out {
2018-04-10 09:46:16.279    "Battery" : 255,
2018-04-10 09:46:16.292    "RSSI" : 12,
2018-04-10 09:46:16.305    "description" : "",
2018-04-10 09:46:16.318    "dtype" : "Color Switch",
2018-04-10 09:46:16.330    "id" : "000003EB",
2018-04-10 09:46:16.342    "idx" : 8,
2018-04-10 09:46:16.353    "name" : "Group Living Room",
2018-04-10 09:46:16.365    "nvalue" : 10,
2018-04-10 09:46:16.379    "stype" : "RGBWW",
2018-04-10 09:46:16.391    "svalue1" : "98",
2018-04-10 09:46:16.403    "switchType" : "On/Off",
2018-04-10 09:46:16.414    "unit" : 1
2018-04-10 09:46:16.426 }
2018-04-10 09:46:16.438
2018-04-10 09:46:26.155 domoticz/out {
2018-04-10 09:46:26.167    "Battery" : 255,
2018-04-10 09:46:26.179    "Color" : "{\"b\":31,\"cw\":0,\"g\":67,\"m\":3,\"r\":254,\"t\":0,\"ww\":0}",
2018-04-10 09:46:26.193    "Level" : 98,
2018-04-10 09:46:26.205    "RSSI" : 12,
2018-04-10 09:46:26.217    "description" : "",
2018-04-10 09:46:26.229    "dtype" : "Color Switch",
2018-04-10 09:46:26.242    "id" : "00000002",
2018-04-10 09:46:26.254    "idx" : 2,
2018-04-10 09:46:26.266    "name" : "Hue color spot 1",
2018-04-10 09:46:26.278    "nvalue" : 10,
2018-04-10 09:46:26.290    "stype" : "RGBWW",
2018-04-10 09:46:26.303    "svalue1" : "98",
2018-04-10 09:46:26.317    "switchType" : "Dimmer",
2018-04-10 09:46:26.329    "unit" : 1
2018-04-10 09:46:26.341 }
2018-04-10 09:46:26.353
2018-04-10 09:46:26.365 domoticz/out {
2018-04-10 09:46:26.378    "Battery" : 255,
2018-04-10 09:46:26.390    "RSSI" : 12,
2018-04-10 09:46:26.403    "description" : "",
2018-04-10 09:46:26.415    "dtype" : "Color Switch",
2018-04-10 09:46:26.427    "id" : "000003EB",
2018-04-10 09:46:26.439    "idx" : 8,
2018-04-10 09:46:26.451    "name" : "Group Living Room",
2018-04-10 09:46:26.463    "nvalue" : 10,
2018-04-10 09:46:26.475    "stype" : "RGBWW",
2018-04-10 09:46:26.487    "svalue1" : "98",
2018-04-10 09:46:26.499    "switchType" : "On/Off",
2018-04-10 09:46:26.510    "unit" : 1
2018-04-10 09:46:26.522 }
2018-04-10 09:46:26.534

2018-04-10 09:46:56.344 domoticz/out {
2018-04-10 09:46:56.356    "Battery" : 255,
2018-04-10 09:46:56.369    "Color" : "{\"b\":69,\"cw\":0,\"g\":254,\"m\":3,\"r\":234,\"t\":0,\"ww\":0}",
2018-04-10 09:46:56.381    "Level" : 98,
2018-04-10 09:46:56.393    "RSSI" : 12,
2018-04-10 09:46:56.405    "description" : "",
2018-04-10 09:46:56.417    "dtype" : "Color Switch",
2018-04-10 09:46:56.429    "id" : "00000002",
2018-04-10 09:46:56.441    "idx" : 2,
2018-04-10 09:46:56.453    "name" : "Hue color spot 1",
2018-04-10 09:46:56.465    "nvalue" : 10,
2018-04-10 09:46:56.477    "stype" : "RGBWW",
2018-04-10 09:46:56.492    "svalue1" : "98",
2018-04-10 09:46:56.504    "switchType" : "Dimmer",
2018-04-10 09:46:56.516    "unit" : 1
2018-04-10 09:46:56.528 }
2018-04-10 09:46:56.540
2018-04-10 09:46:56.552 domoticz/out {
2018-04-10 09:46:56.564    "Battery" : 255,
2018-04-10 09:46:56.576    "RSSI" : 12,
2018-04-10 09:46:56.588    "description" : "",
2018-04-10 09:46:56.600    "dtype" : "Color Switch",
2018-04-10 09:46:56.612    "id" : "000003EB",
2018-04-10 09:46:56.624    "idx" : 8,
2018-04-10 09:46:56.635    "name" : "Group Living Room",
2018-04-10 09:46:56.647    "nvalue" : 10,
2018-04-10 09:46:56.659    "stype" : "RGBWW",
2018-04-10 09:46:56.672    "svalue1" : "98",
2018-04-10 09:46:56.684    "switchType" : "On/Off",
2018-04-10 09:46:56.695    "unit" : 1
2018-04-10 09:46:56.708 }
2018-04-10 09:46:56.720
2018-04-10 09:47:06.431 domoticz/out {
2018-04-10 09:47:06.445    "Battery" : 255,
2018-04-10 09:47:06.457    "Color" : "{\"b\":69,\"cw\":0,\"g\":254,\"m\":3,\"r\":235,\"t\":0,\"ww\":0}",
2018-04-10 09:47:06.469    "Level" : 98,
2018-04-10 09:47:06.481    "RSSI" : 12,
2018-04-10 09:47:06.493    "description" : "",
2018-04-10 09:47:06.505    "dtype" : "Color Switch",
2018-04-10 09:47:06.518    "id" : "00000002",
2018-04-10 09:47:06.530    "idx" : 2,
2018-04-10 09:47:06.542    "name" : "Hue color spot 1",
2018-04-10 09:47:06.555    "nvalue" : 10,
2018-04-10 09:47:06.567    "stype" : "RGBWW",
2018-04-10 09:47:06.579    "svalue1" : "98",
2018-04-10 09:47:06.591    "switchType" : "Dimmer",
2018-04-10 09:47:06.603    "unit" : 1
2018-04-10 09:47:06.615 }
2018-04-10 09:47:06.627
2018-04-10 09:47:06.639 domoticz/out {
2018-04-10 09:47:06.651    "Battery" : 255,
2018-04-10 09:47:06.663    "RSSI" : 12,
2018-04-10 09:47:06.675    "description" : "",
2018-04-10 09:47:06.688    "dtype" : "Color Switch",
2018-04-10 09:47:06.700    "id" : "000003EB",
2018-04-10 09:47:06.712    "idx" : 8,
2018-04-10 09:47:06.724    "name" : "Group Living Room",
2018-04-10 09:47:06.736    "nvalue" : 10,
2018-04-10 09:47:06.748    "stype" : "RGBWW",
2018-04-10 09:47:06.760    "svalue1" : "98",
2018-04-10 09:47:06.772    "switchType" : "On/Off",
2018-04-10 09:47:06.785    "unit" : 1
2018-04-10 09:47:06.797 }
2018-04-10 09:47:06.809

2018-04-10 09:48:37.228 domoticz/out {
2018-04-10 09:48:37.241    "Battery" : 255,
2018-04-10 09:48:37.254    "Color" : "{\"b\":159,\"cw\":0,\"g\":0,\"m\":3,\"r\":224,\"t\":0,\"ww\":0}",
2018-04-10 09:48:37.267    "Level" : 98,
2018-04-10 09:48:37.279    "RSSI" : 12,
2018-04-10 09:48:37.291    "description" : "",
2018-04-10 09:48:37.303    "dtype" : "Color Switch",
2018-04-10 09:48:37.314    "id" : "00000002",
2018-04-10 09:48:37.326    "idx" : 2,
2018-04-10 09:48:37.338    "name" : "Hue color spot 1",
2018-04-10 09:48:37.349    "nvalue" : 10,
2018-04-10 09:48:37.361    "stype" : "RGBWW",
2018-04-10 09:48:37.374    "svalue1" : "98",
2018-04-10 09:48:37.386    "switchType" : "Dimmer",
2018-04-10 09:48:37.398    "unit" : 1
2018-04-10 09:48:37.410 }
2018-04-10 09:48:37.423
2018-04-10 09:48:37.434 domoticz/out {
2018-04-10 09:48:37.446    "Battery" : 255,
2018-04-10 09:48:37.458    "RSSI" : 12,
2018-04-10 09:48:37.470    "description" : "",
2018-04-10 09:48:37.487    "dtype" : "Color Switch",
2018-04-10 09:48:37.503    "id" : "000003EB",
2018-04-10 09:48:37.515    "idx" : 8,
2018-04-10 09:48:37.527    "name" : "Group Living Room",
2018-04-10 09:48:37.541    "nvalue" : 10,
2018-04-10 09:48:37.553    "stype" : "RGBWW",
2018-04-10 09:48:37.566    "svalue1" : "98",
2018-04-10 09:48:37.578    "switchType" : "On/Off",
2018-04-10 09:48:37.591    "unit" : 1
2018-04-10 09:48:37.603 }
2018-04-10 09:48:37.615
2018-04-10 09:48:47.345 domoticz/out {
2018-04-10 09:48:47.359    "Battery" : 255,
2018-04-10 09:48:47.371    "Color" : "{\"b\":159,\"cw\":0,\"g\":0,\"m\":3,\"r\":224,\"t\":0,\"ww\":0}",
2018-04-10 09:48:47.384    "Level" : 98,
2018-04-10 09:48:47.396    "RSSI" : 12,
2018-04-10 09:48:47.408    "description" : "",
2018-04-10 09:48:47.420    "dtype" : "Color Switch",
2018-04-10 09:48:47.433    "id" : "00000002",
2018-04-10 09:48:47.445    "idx" : 2,
2018-04-10 09:48:47.456    "name" : "Hue color spot 1",
2018-04-10 09:48:47.468    "nvalue" : 10,
2018-04-10 09:48:47.480    "stype" : "RGBWW",
2018-04-10 09:48:47.492    "svalue1" : "98",
2018-04-10 09:48:47.504    "switchType" : "Dimmer",
2018-04-10 09:48:47.519    "unit" : 1
2018-04-10 09:48:47.533 }
2018-04-10 09:48:47.547
2018-04-10 09:48:47.559 domoticz/out {
2018-04-10 09:48:47.570    "Battery" : 255,
2018-04-10 09:48:47.582    "RSSI" : 12,
2018-04-10 09:48:47.595    "description" : "",
2018-04-10 09:48:47.607    "dtype" : "Color Switch",
2018-04-10 09:48:47.619    "id" : "000003EB",
2018-04-10 09:48:47.631    "idx" : 8,
2018-04-10 09:48:47.643    "name" : "Group Living Room",
2018-04-10 09:48:47.655    "nvalue" : 10,
2018-04-10 09:48:47.667    "stype" : "RGBWW",
2018-04-10 09:48:47.679    "svalue1" : "98",
2018-04-10 09:48:47.691    "switchType" : "On/Off",
2018-04-10 09:48:47.704    "unit" : 1
2018-04-10 09:48:47.716 }
2018-04-10 09:48:47.728

```

**Comment by emontnemery** [comment]
@jonferreira Is everything else working,except the double notifications?
**Regarding the double notifications:**
Can you paste the full log to make more clear what triggers the notifications?
Do you get double notification always, or only in some case, for example only when using Domoticz Web UI?
Did you not see double notifications in the previous build?

**Comment by jonferreira** [comment]
yes everything seems to work, just these double notifications that I've never seen before (well there was that *one* time that I mentioned yesterday but could never reproduce!)

I only get double notifications when I change color regardless if it's via Domoticz or Hue App - wondering if it has something to do with xy conversion as sometimes I do see slighly changes on the Color variable

2018-04-10 09:46:56.369    "Color" : "{\"b\":69,\"cw\":0,\"g\":254,\"m\":3,\"r\":**234**,\"t\":0,\"ww\":0}"

vs

2018-04-10 09:47:06.457    "Color" : "{\"b\":69,\"cw\":0,\"g\":254,\"m\":3,\"r\":**235**,\"t\":0,\"ww\":0}"

[log01.log](https://github.com/domoticz/domoticz/files/1894168/log01.log)

**Comment by emontnemery** [comment]
Yeah, I agree it's probably related to rounding of the values.
Please do some more testing with http://montnemery.com/domoticz_hue/linux_armv7l/008/domoticz_linux_armv7l.tgz, some additional prints to confirm that's the case.

If confirmed, I'm thinking about blocking the notification if the change is, say, less than 1%.
Do you think that's a reasonable approach?

**Comment by jonferreira** [comment]
OK tested again 

```
2018-04-11 12:55:25.702 domoticz/out {
2018-04-11 12:55:25.715    "Battery" : 255,
2018-04-11 12:55:25.727    "Color" : "{\"b\":69,\"cw\":0,\"g\":254,\"m\":3,\"r\":234,\"t\":0,\"ww\":0}",
2018-04-11 12:55:25.740    "Level" : 99,
2018-04-11 12:55:25.752    "RSSI" : 12,
2018-04-11 12:55:25.765    "description" : "",
2018-04-11 12:55:25.777    "dtype" : "Color Switch",
2018-04-11 12:55:25.789    "id" : "00000002",
2018-04-11 12:55:25.800    "idx" : 2,
2018-04-11 12:55:25.812    "name" : "Hue color spot 1",
2018-04-11 12:55:25.824    "nvalue" : 10,
2018-04-11 12:55:25.835    "stype" : "RGBWW",
2018-04-11 12:55:25.847    "svalue1" : "99",
2018-04-11 12:55:25.859    "switchType" : "Dimmer",
2018-04-11 12:55:25.871    "unit" : 1
2018-04-11 12:55:25.883 }
2018-04-11 12:55:25.894
2018-04-11 12:55:25.907 domoticz/out {
2018-04-11 12:55:25.920    "Battery" : 255,
2018-04-11 12:55:25.932    "RSSI" : 12,
2018-04-11 12:55:25.944    "description" : "",
2018-04-11 12:55:25.956    "dtype" : "Color Switch",
2018-04-11 12:55:25.969    "id" : "000003EB",
2018-04-11 12:55:25.982    "idx" : 8,
2018-04-11 12:55:25.994    "name" : "Group Living Room",
2018-04-11 12:55:26.006    "nvalue" : 10,
2018-04-11 12:55:26.018    "stype" : "RGBWW",
2018-04-11 12:55:26.030    "svalue1" : "99",
2018-04-11 12:55:26.042    "switchType" : "On/Off",
2018-04-11 12:55:26.054    "unit" : 1
2018-04-11 12:55:26.066 }
2018-04-11 12:55:26.078
2018-04-11 12:55:35.783 domoticz/out {
2018-04-11 12:55:35.796    "Battery" : 255,
2018-04-11 12:55:35.808    "Color" : "{\"b\":69,\"cw\":0,\"g\":254,\"m\":3,\"r\":235,\"t\":0,\"ww\":0}",
2018-04-11 12:55:35.820    "Level" : 99,
2018-04-11 12:55:35.832    "RSSI" : 12,
2018-04-11 12:55:35.844    "description" : "",
2018-04-11 12:55:35.856    "dtype" : "Color Switch",
2018-04-11 12:55:35.868    "id" : "00000002",
2018-04-11 12:55:35.880    "idx" : 2,
2018-04-11 12:55:35.892    "name" : "Hue color spot 1",
2018-04-11 12:55:35.904    "nvalue" : 10,
2018-04-11 12:55:35.915    "stype" : "RGBWW",
2018-04-11 12:55:35.928    "svalue1" : "99",
2018-04-11 12:55:35.941    "switchType" : "Dimmer",
2018-04-11 12:55:35.952    "unit" : 1
2018-04-11 12:55:35.964 }
2018-04-11 12:55:35.976
2018-04-11 12:55:35.988 domoticz/out {
2018-04-11 12:55:36.000    "Battery" : 255,
2018-04-11 12:55:36.012    "RSSI" : 12,
2018-04-11 12:55:36.024    "description" : "",
2018-04-11 12:55:36.039    "dtype" : "Color Switch",
2018-04-11 12:55:36.051    "id" : "000003EB",
2018-04-11 12:55:36.063    "idx" : 8,
2018-04-11 12:55:36.076    "name" : "Group Living Room",
2018-04-11 12:55:36.087    "nvalue" : 10,
2018-04-11 12:55:36.100    "stype" : "RGBWW",
2018-04-11 12:55:36.111    "svalue1" : "99",
2018-04-11 12:55:36.123    "switchType" : "On/Off",
2018-04-11 12:55:36.135    "unit" : 1
2018-04-11 12:55:36.148 }
2018-04-11 12:55:36.159
```

[log01.log](https://github.com/domoticz/domoticz/files/1898664/log01.log)

It seems like the second poll returns a slightly different XY value than the previous one.

**Comment by jonferreira** [comment]
It's a hard call really, technically the color have changed (i assume) although it's such a slight change that you probably won't **see** it. So under that light there's no bug to fix, it's just how Hue works I guess.

Although it may be worth to add a threshold in color change for a new MQTT notification like you suggested.

Maybe truncate the XY values to 3 decimal places, convert to RGBW and if it ends up with a different RGB value send notification?

**Comment by emontnemery** [comment]
OK, new version with hysteresis on brightness, color values and color temperature:
http://montnemery.com/domoticz_hue/linux_armv7l/009/domoticz_linux_armv7l.tgz

It spams the log quite a lot to help debugging in case the hysteresis doesn't work.

By the way, how come your groups has switchType set to On/Off, have you manually changed it?

**Comment by jonferreira** [comment]
> By the way, how come your groups has switchType set to On/Off, have you manually changed it?
Never changed anything, wouldn't even know how to do it. This is a test instance that I've set up just for this and the DB was created by the first Beta you gave me. I've now deleted all the devices, polled the Hue bridge again (which recreated the devices) and it now shows up as Dimmer

That said, something isn't right now. 

Changes done on the HUE app doesn't update MQTT or Domoticz at all
Color change via Domoticz does send an MQTT notification but twice, even though the Color variable doesn't seem to have changed

```
2018-04-12 16:24:23.354 domoticz/out {
2018-04-12 16:24:23.365    "Battery" : 255,
2018-04-12 16:24:23.378    "Color" : "{\"b\":41,\"cw\":0,\"g\":18,\"m\":3,\"r\":255,\"t\":0,\"ww\":0}",
2018-04-12 16:24:23.390    "Level" : 100,
2018-04-12 16:24:23.404    "RSSI" : 12,
2018-04-12 16:24:23.416    "description" : "",
2018-04-12 16:24:23.431    "dtype" : "Color Switch",
2018-04-12 16:24:23.443    "id" : "00000002",
2018-04-12 16:24:23.455    "idx" : 2,
2018-04-12 16:24:23.476    "name" : "Hue color spot 1",
2018-04-12 16:24:23.490    "nvalue" : 10,
2018-04-12 16:24:23.503    "stype" : "RGBWW",
2018-04-12 16:24:23.515    "svalue1" : "100",
2018-04-12 16:24:23.527    "switchType" : "Dimmer",
2018-04-12 16:24:23.538    "unit" : 1
2018-04-12 16:24:23.552 }
2018-04-12 16:24:23.563
2018-04-12 16:24:30.050 domoticz/out {
2018-04-12 16:24:30.065    "Battery" : 255,
2018-04-12 16:24:30.090    "Color" : "{\"b\":41,\"cw\":0,\"g\":18,\"m\":3,\"r\":255,\"t\":0,\"ww\":0}",
2018-04-12 16:24:30.105    "Level" : 100,
2018-04-12 16:24:30.118    "RSSI" : 12,
2018-04-12 16:24:30.134    "description" : "",
2018-04-12 16:24:30.146    "dtype" : "Color Switch",
2018-04-12 16:24:30.174    "id" : "00000002",
2018-04-12 16:24:30.187    "idx" : 2,
2018-04-12 16:24:30.202    "name" : "Hue color spot 1",
2018-04-12 16:24:30.214    "nvalue" : 10,
2018-04-12 16:24:30.227    "stype" : "RGBWW",
2018-04-12 16:24:30.252    "svalue1" : "100",
2018-04-12 16:24:30.265    "switchType" : "Dimmer",
2018-04-12 16:24:30.278    "unit" : 1
2018-04-12 16:24:30.291 }
2018-04-12 16:24:30.316
2018-04-12 16:24:30.329 domoticz/out {
2018-04-12 16:24:30.342    "Battery" : 255,
2018-04-12 16:24:30.354    "Color" : "{\"b\":41,\"cw\":0,\"g\":18,\"m\":3,\"r\":255,\"t\":0,\"ww\":0}",
2018-04-12 16:24:30.380    "Level" : 100,
2018-04-12 16:24:30.393    "RSSI" : 12,
2018-04-12 16:24:30.405    "description" : "",
2018-04-12 16:24:30.417    "dtype" : "Color Switch",
2018-04-12 16:24:30.430    "id" : "000003EB",
2018-04-12 16:24:30.443    "idx" : 8,
2018-04-12 16:24:30.459    "name" : "Group Living Room",
2018-04-12 16:24:30.477    "nvalue" : 10,
2018-04-12 16:24:30.490    "stype" : "RGBWW",
2018-04-12 16:24:30.502    "svalue1" : "100",
2018-04-12 16:24:30.515    "switchType" : "Dimmer",
2018-04-12 16:24:30.527    "unit" : 1
2018-04-12 16:24:30.540 }
```
[log01.log](https://github.com/domoticz/domoticz/files/1903840/log01.log)

**Comment by emontnemery** [comment]
Thanks for the log, I hope I found the root cause for still getting double notifications when changing colors from Domoticz UI as well as missed notifications when switching on/off a light from external apps.
Hopefully fixed here: http://montnemery.com/domoticz_hue/linux_armv7l/010/domoticz_linux_armv7l.tgz

Do you also see missed notification when changing color or color temperature from the hue app?

If you do, please point to somewhere in the log where light state has changed, but it's incorrectly suppressed:
```
2018-04-12 16:24:09.898  PhilipsHue::StatesSimilar s1: on: 1, bri: 100, mode: 1, hue: 20392, sat: 247, ct: 0, x: 0.408400, y: 0.516800
2018-04-12 16:24:09.899  PhilipsHue::StatesSimilar s2: on: 1, bri: 100, mode: 1, hue: 20392, sat: 247, ct: 0, x: 0.460800, y: 0.475300
2018-04-12 16:24:09.899  PhilipsHue::StatesSimilar res: 1
```
(s1 is old state, s2 is new state, if res is 1 there won't be a notification)

Regarding the groups:
Maybe i unintentionally changed something to make them appear as RGB dimmer instead of On/Off. Do you think it makes more sense for them to appear as On/Off or Dimmer in Domoticz?

PS:
Sorry about the many iterations, I don't own genuine Hue bulbs so I can't really test the changes..

**Comment by jonferreira** [comment]
>Do you also see missed notification when changing color or color temperature from the hue app?

I did yes but seems to be OK now

>Do you think it makes more sense for them to appear as On/Off or Dimmer in Domoticz?

Dimmer makes more sense. Groups in the Hue app works exactly the same way as a single bulb so you can turn on / off / dimm and change color as well

Actually been meaning to ask, how hard would be to add an option on the Hue "hardware" to enable / disable import of bulbs / groups / scenes (3 checkboxes perhaps)? I get spammed with all these useless scenes that I'll never need but keeps being created in domo...

> Sorry about the many iterations, I don't own genuine Hue bulbs so I can't really test the changes..

No worries! Seems fine now though!

```

2018-04-13 11:25:00.631 domoticz/out {
2018-04-13 11:25:00.652    "Battery" : 255,
2018-04-13 11:25:00.664    "Color" : "{\"b\":45,\"cw\":0,\"g\":66,\"m\":3,\"r\":254,\"t\":0,\"ww\":0}",
2018-04-13 11:25:00.676    "Level" : 99,
2018-04-13 11:25:00.688    "RSSI" : 12,
2018-04-13 11:25:00.700    "description" : "",
2018-04-13 11:25:00.712    "dtype" : "Color Switch",
2018-04-13 11:25:00.724    "id" : "00000002",
2018-04-13 11:25:00.736    "idx" : 2,
2018-04-13 11:25:00.748    "name" : "Hue color spot 1",
2018-04-13 11:25:00.759    "nvalue" : 10,
2018-04-13 11:25:00.772    "stype" : "RGBWW",
2018-04-13 11:25:00.783    "svalue1" : "99",
2018-04-13 11:25:00.796    "switchType" : "Dimmer",
2018-04-13 11:25:00.808    "unit" : 1
2018-04-13 11:25:00.820 }
2018-04-13 11:25:00.832
2018-04-13 11:25:00.844 domoticz/out {
2018-04-13 11:25:00.856    "Battery" : 255,
2018-04-13 11:25:00.868    "Color" : "{\"b\":45,\"cw\":0,\"g\":66,\"m\":3,\"r\":254,\"t\":0,\"ww\":0}",
2018-04-13 11:25:00.880    "Level" : 99,
2018-04-13 11:25:00.892    "RSSI" : 12,
2018-04-13 11:25:00.906    "description" : "",
2018-04-13 11:25:00.918    "dtype" : "Color Switch",
2018-04-13 11:25:00.930    "id" : "000003EB",
2018-04-13 11:25:00.942    "idx" : 8,
2018-04-13 11:25:00.954    "name" : "Group Living Room",
2018-04-13 11:25:00.966    "nvalue" : 10,
2018-04-13 11:25:00.978    "stype" : "RGBWW",
2018-04-13 11:25:00.990    "svalue1" : "99",
2018-04-13 11:25:01.003    "switchType" : "Dimmer",
2018-04-13 11:25:01.015    "unit" : 1
2018-04-13 11:25:01.027 }
2018-04-13 11:25:01.038
```

**Comment by emontnemery** [comment]
@jonferreira So, long story short, everything is now working as expected and this should be ready for merge (after removing all the extra prints)?
Perfectly doable to add options to not import groups and scenes, please open a separate issue for that.

**Comment by jonferreira** [comment]
Yes seems to be ready for general availability

**Comment by emontnemery** [comment]
OK, rebased to latest development + removed all the extra prints.
Do you mind do a final test to make sure everything is still working: http://montnemery.com/domoticz_hue/linux_armv7l/011/domoticz_linux_armv7l.tgz

**Comment by dotms** [comment]
I'll try and see if I can do some testing. My current build (of 7 days ago) does not register (any) actions done through the hue app or physical hue remotes, not even a power on/off. The other way around (controlling from Domoticz) works fine though.

Edit: Just did some quick testing using your build on a RPi2. Events from the Hue bridge are coming in again. That's great! 

One thing that I did notice: if I dim a group the dim level of the group in Domoticz rises/drops accordingly. The individual bulbs remain at their initial position though (until the group is powered off and then back on). Is that intentional?

**Comment by emontnemery** [comment]
@dotms Not intentional. It could be a bug, but could also be due to how the Hue bridge reports individual bulb status when dimming groups.
Try the last build with extra prints, http://montnemery.com/domoticz_hue/linux_armv7l/010/domoticz_linux_armv7l.tgz, and collect log.
You can for example test like this:
- set one bulb to fully red
- wait 30 s
- set another bulb to fully green
- wait 30s
- dim a group which includes both bulbs
- wait 30s.
Then paste the log here and let's see.

**Comment by dotms** [comment]
@emontnemery Is there anything special that I should do to get some meaningful logs out of it? I tried running it with -loglevel 0 and -verbose 1 but all I get are just the regular events like "2018-04-15 17:20:27.313 (hue) Color Switch (Vaaslamp onder)".

edit: nevermind, found it. -loglevel 3 and -debug did the trick.

edit2: see log at https://www.dropbox.com/s/px9bewlnq4zn9z3/domoticz.log?dl=0 . What I did:
- Set 'Vaaslamp onder' to red (from Domoticz)
- Set 'Vaaslamp boven' to blue (from Domoticz)
- Set 'Group Woonkamer' to 50% (from Hue app). This group contains both these bulbs.

**Comment by emontnemery** [comment]
@dotms Did you change to this build: http://montnemery.com/domoticz_hue/linux_armv7l/010/domoticz_linux_armv7l.tgz, there should be a lot of prints like this in the log, but there are none:
```
2018-04-12 16:24:09.898  PhilipsHue::StatesSimilar s1: on: 1, bri: 100, mode: 1, hue: 20392, sat: 247, ct: 0, x: 0.408400, y: 0.516800
2018-04-12 16:24:09.899  PhilipsHue::StatesSimilar s2: on: 1, bri: 100, mode: 1, hue: 20392, sat: 247, ct: 0, x: 0.460800, y: 0.475300
2018-04-12 16:24:09.899  PhilipsHue::StatesSimilar res: 1
```

**Comment by dotms** [comment]
I don't know what happened but I redownloaded the zip and now I see many of the lines you mention in the console. New log: https://www.dropbox.com/s/px9bewlnq4zn9z3/domoticz.log?dl=0

Some things that I did (all from Domoticz this time)
- Set 'Vaaslamp boven' to red
- Set 'Vaaslamp onder' to blue
- Set 'Group Woonkamer' to 25% dimming
- Turned off 'Group Woonkamer'

Note that both the 'Vaaslamp' bulbs received their dimming level as soon as I turned off the 'Group Woonkamer', but before that action they didn't show as dimmed in Domoticz.

**Comment by emontnemery** [comment]
Set group (with device ID 21) to 25%:
`2018-04-15 18:56:00.985  MAIN SwitchLight idx:21 cmd:Set Color lvl:26`
Devices 3, 4, 5, 6, 7, 8 updated to 25% after poll of Hue bridge:
```
2018-04-15 18:56:12.042  SQLR result: 3;Hoek beneden voor;
2018-04-15 18:56:12.046  SQLQ query : UPDATE DeviceStatus SET LastLevel='25' WHERE (ID = 3)
2018-04-15 18:56:12.233  SQLR result: 4;Vaaslamp boven;
2018-04-15 18:56:12.237  SQLQ query : UPDATE DeviceStatus SET LastLevel='25' WHERE (ID = 4)
2018-04-15 18:56:12.415  SQLR result: 5;Vaaslamp onder;
2018-04-15 18:56:12.420  SQLQ query : UPDATE DeviceStatus SET LastLevel='25' WHERE (ID = 5)
2018-04-15 18:56:12.513  SQLR result: 6;Schemerlamp;
2018-04-15 18:56:12.517  SQLQ query : UPDATE DeviceStatus SET LastLevel='25' WHERE (ID = 6)
2018-04-15 18:56:12.610  SQLR result: 7;Tafellampje;
2018-04-15 18:56:12.614  SQLQ query : UPDATE DeviceStatus SET LastLevel='25' WHERE (ID = 7)
2018-04-15 18:56:12.706  SQLR result: 8;Sfeerlamp woonkamer achter;
2018-04-15 18:56:12.710  SQLQ query : UPDATE DeviceStatus SET LastLevel='25' WHERE (ID = 8)
```
Switch the group off:
`2018-04-15 18:56:37.184  WEBH : Host:192.168.177.102 Uri;/json.htm?type=command&param=switchlight&idx=21&switchcmd=Off&level=0&passcode=`

The Web UI is based on periodic polling of the state, like this:
2018-04-15 18:56:38.411  WEBH : Host:192.168.177.102 Uri;/json.htm?type=devices&filter=light&used=true&order=[Order]&lastupdate=1523811315&plan=0

However, no poll happens from the time you set the group to 25% until you switch the group off, hence no update of the UI. This might be a WebUI bug, not related to Philips Hue.

To verify this, try to:
- Set the group to 25%
- Reload the switches page after at least 30 s (to make sure the Hue Bridge has been polled)
- Do you see the correct level now?

**Comment by dotms** [comment]
Just verified this. I changed the dimming level of the group. After a Hue polling cycle and a web UI polling cycle the individual bulbs show the same percentage as the group was set to. So the percentage not showing correctly earlier was more a browser or web UI issue than a Hue plugin issue :)

Note: Possibly due to the way Hue works I notice that an additional polling cycle is needed to process the state of the bulbs after a change is made to the group. The group is updated the next cycle, the bulbs the cycle thereafter.

**Comment by jonferreira** [comment]
Yes sounds like "normal" behaviour @dotms 

@emontnemery seems ready to be released in my opinion, thanks for all the hard work!
